### ISSUE TITLE
Use the network setting for max TTL extension

### ISSUE THREAD

**Issue by dmkozh** [stellar/stellar-cli#1204]
During TTL extension operation [ledgers_to_extend](https://github.com/stellar/soroban-cli/blob/4febbe2f02faef4f73422179f2c925050bf2bda5/cmd/soroban-cli/src/commands/contract/extend.rs#L94) is incorrectly truncated to `MAX_LEDGERS_TO_EXTEND` constant. Instead, the max ledgers to extend should be read from the network config, specifically from the `stateArchivalSettings.maxEntryTTL` setting for the respective network. The check could probably happen on the simulation side, so it could be sufficient for CLI to simply remove the truncation code.

**Comment by willemneal** [comment]
@dmkozh Where is this network endpoint? Is it through Horizon? Currently we only use soroban-rpc. Do you suggest this should be a compile time constant?

**Comment by dmkozh** [comment]
No, this is not a constant, which is the reason for the issue. You need to fetch the `stateArchivalSettings` ledger entry and use `maxEntryTTL` from it. I'm not sure if RPC provides a shortcut for that, but if you just use the `getLedgerEntry` API using a ledger key, then your key would be of type `CONFIG_SETTING` with `configSettingID` set to `CONFIG_SETTING_STATE_ARCHIVAL`.
